OSDOCS#10143: Updating that SA API token secrets are no longer automa… (#75196)
openshift/openshift-docs: modules/service-account-auto-secret-removed.adoc

:_mod-docs-content-type: CONCEPT
[id="auto-generated-sa-token-secrets_{context}"]
= Automatically generated image pull secrets

> By default, {product-title} creates a docker configuration image pull secret for each service account.

Reviewer: ```suggestion
By default, {product-title} creates an image pull secret for each service account.
```

[NOTE]
====
Prior to {product-title} 4.16, a long-lived service account API token secret was also generated for each service account that was created. Starting with {product-title} 4.16, this service account API token secret is no longer created.

After upgrading to {product-version}, any existing long-lived service account API token secrets are not deleted and will continue to function. For information about detecting long-lived API tokens that are in use in your cluster or deleting them if they are not needed, see the Red Hat Knowledgebase article link:https://access.redhat.com/articles/7058801[Long-lived service account API tokens in OpenShift Container Platform].
====

This docker configuration image pull secret is necessary to integrate the {product-registry} into the cluster's user authentication and authorization system.

> However, if you do not enable the `ImageRegistry` capability or if you disable the integrated {product-registry} in the Cluster Image Registry Operator's configuration, a dockercfg secret is not generated for each service account.

Reviewer: Add something such as: When the integrated {product-registry} is disabled on a cluster that previously had it enabled, the previously auto-generated image pull secrets will be deleted automatically.

[id="auto-generated-sa-token-secrets_{context}"]
= Automatically generated secrets

By default, {product-title} creates a docker configuration image pull secret for each service account.

[NOTE]
====
Prior to {product-title} 4.16, a long-lived service account API token secret was also generated for each service account that was created. Starting with {product-title} 4.16, this service account API token secret is no longer created.

> After upgrading to {product-version}, any existing long-lived service account API token secrets are not deleted and will continue to function. For information about detecting long-lived API tokens that are in use in your cluster or deleting them if they are not needed, see the Red Hat Knowledgebase article link:https://access.redhat.com/articles/7058801[Long-lived service account API tokens in OpenShift Container Platform].

Reviewer: replace **long-lived** with **legacy**

Reviewer: > ~~replace **long-lived** with **legacy**~~
actually, that terminology is ok.

====

This docker configuration image pull secret is necessary to integrate the {product-registry} into the cluster's user authentication and authorization system.

However, if you do not enable the `ImageRegistry` capability or if you disable the integrated {product-registry} in the Cluster Image Registry Operator's configuration, a dockercfg secret is not generated for each service account.

> [WARNING]

Reviewer: This warning section can be removed. It is referencing  the legacy API token secrets.

Author: @sanchezl So just checking - we don't need to warn people away from using the dockercfg image pull secrets for their own use?

Reviewer: Correct. No need to warn them about the image pull secrets.
